src/SFA.DAS.EmployerIncentives.Data/ApprenticeshipIncentives/ApprenticeshipIncentiveDataRepository.cs

```diff
@@ -151,6 +129,34 @@ public async Task Delete(ApprenticeshipIncentiveModel apprenticeshipIncentive)
             }
         }
 
+        private async Task<ApprenticeshipIncentiveModel> Get(Expression<Func<ApprenticeshipIncentive, bool>> predicate)
+        {
+            var apprenticeshipIncentive = await _dbContext.ApprenticeshipIncentives.FirstOrDefaultAsync(predicate);
+
+            if (apprenticeshipIncentive != null)
+            {
+                await PopulateApprenticeshipIncentive(apprenticeshipIncentive);
+                return apprenticeshipIncentive?.Map(_dbContext.CollectionPeriods.AsEnumerable());
+            }
+
+            return null;
+        }
+
+        private async Task PopulateApprenticeshipIncentive(ApprenticeshipIncentive apprenticeshipIncentive)
+        {
+            apprenticeshipIncentive.PendingPayments = await _dbContext.PendingPayments.Where(x => x.ApprenticeshipIncentiveId == apprenticeshipIncentive.Id).ToListAsync();
+            foreach (var pendingPayment in apprenticeshipIncentive.PendingPayments)
```

Comment:
foreach loop might be slower and more expensive i.e. number of round trip to SQL server etc compare to doing include, alternatively use an "in" statement (IN statement does have a limit of 2000 or something so I would check actual data and do max per apprenticeshipIncentive and based on that chose one or the other)

Reply:
Typically there are 2 pending payments per incentive, there can be more than that if payments are clawed back and recalculated, or less if earnings removed, but those are outliers

Reply:
This has been changed to fetch the pending payments and validation results in the initial fetch, as per Kevin's suggestion